atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/expectations/ExpectationsMeasuringUtils.java

```diff
@@ -0,0 +1,101 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.impl.expectations;
+
+import com.google.common.collect.Multimap;
+import com.palantir.atlasdb.keyvalue.api.RangeRequest;
+import com.palantir.atlasdb.keyvalue.api.RowResult;
+import com.palantir.atlasdb.keyvalue.api.Value;
+import com.palantir.atlasdb.util.Measurable;
+import com.palantir.util.paging.TokenBackedBasicResultsPage;
+import java.lang.reflect.Array;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.function.ToLongFunction;
+
+public final class ExpectationsMeasuringUtils {
+    private ExpectationsMeasuringUtils() {}
+
+    public static long toLongSizeInBytes(Map<? extends Measurable, Long> map) {
+        return sizeInBytes(map, unused -> Long.BYTES);
+    }
+
+    public static long toArraySizeInBytes(Map<? extends Measurable, byte[]> map) {
+        return sizeInBytes(map, Array::getLength);
+    }
+
+    public static long byteArraysSizeInBytes(Collection<byte[]> byteArrays) {
+        return sizeInBytes(byteArrays, Array::getLength);
+    }
+
+    public static long setResultSizeInBytes(RowResult<Set<Long>> rowResult) {
+        return sizeInBytes(rowResult, set -> (long) set.size() * Long.BYTES);
+    }
+
+    /**
+     * Ignoring the size of token for next page because the interface method
+     * {@link TokenBackedBasicResultsPage#getTokenForNextPage} might have side effects (e.g. interact with the kvs).
+     * The interface specification is ambiguous.
+     */
+    public static long pageByRequestSizeInBytes(
+            Map<RangeRequest, TokenBackedBasicResultsPage<RowResult<Value>, byte[]>> pageByRange) {
+        return pageByRange.values().stream()
+                .map(TokenBackedBasicResultsPage::getResults)
+                .flatMap(Collection::stream)
+                .mapToLong(ExpectationsMeasuringUtils::sizeInBytes)
+                .sum();
+    }
+
+    public static long sizeInBytes(Multimap<? extends Measurable, Long> map) {
+        return sizeInBytes(map.keys()) + Long.BYTES * ((long) map.size());
+    }
+
+    public static long sizeInBytes(Entry<? extends Measurable, ? extends Measurable> entry) {
+        return entry.getKey().sizeInBytes() + entry.getValue().sizeInBytes();
+    }
+
+    public static long sizeInBytes(Collection<? extends Measurable> collection) {
+        return sizeInBytes(collection, Measurable::sizeInBytes);
+    }
+
+    public static long sizeInBytes(Map<? extends Measurable, ? extends Measurable> map) {
+        return sizeInBytes(map, Measurable::sizeInBytes);
+    }
+
+    public static long sizeInBytes(RowResult<? extends Measurable> rowResult) {
+        return sizeInBytes(rowResult, Measurable::sizeInBytes);
+    }
+
+    private static <T> long sizeInBytes(Collection<T> collection, ToLongFunction<T> measurer) {
```

Comment:
For readability, let's rename these private helper methods to be something different?